From c35703453ef9759da2592842a9d1980286450e8d Mon Sep 17 00:00:00 2001
From: Venkat Chimata <venkat@nearhop.com>
Date: Tue, 22 Sep 2026 13:46:34 +0530
Subject: [PATCH] nl80211: fix netlink message leak in set_ttlm_link_mapping

The message from nlmsg_alloc() was overwritten by nl80211_bss_msg(),
which allocates its own, leaking one page-sized nl_msg on every TTLM
mapping request. Drop the redundant allocation.

Signed-off-by: Venkat Chimata <venkat@nearhop.com>
---
 src/drivers/driver_nl80211.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 758cb17..7a67e8b 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -15956,9 +15956,6 @@ static int wpa_driver_nl80211_set_ttlm_link_mapping(void *priv, enum wpa_driver_
 	int ret;
 
 	wpa_printf(MSG_DEBUG, "nl80211: MLD: set ttlm params");
-	msg = nlmsg_alloc();
-	if (!msg)
-		return -ENOMEM;
 
 	msg = nl80211_bss_msg(bss, 0, NL80211_CMD_SET_TID_TO_LINK_MAPPING);
 	if (!msg) {
-- 
2.34.1

